blocks/library/embed/index.js

```diff
@@ -26,7 +27,33 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
+	for ( const patternIndex in patterns ) {
+		const pattern = new RegExp( '^https?:\/\/' + patterns[ patternIndex ] + '$', 'i' );
+		if ( url.match( pattern ) ) {
+			return true;
+		}
+	}
+	return false;
+};
+
+const findBlock = ( url ) => {
+	const findMatch = ( blocks ) => {
```

Comment:
Why do we define a function here? Couldn't we just assign `const blocks = [ ...common, ...others ]` and collapse the function into the outer function?